Request for Code Review - Weather App Project (Using vanilla JavaScript and Weather API)

rloterh opened the pull request:
- Data is being fetched from the weather API correctly.
- Displays weather info: Temperature(Celcius/ Fahrenheit), Humidity and Wind speed
- Test cases are passing correctly

rloterh committed:
Setup project structure

rloterh committed:
Create a card to dispaly weather details

rloterh committed:
Create an element component for variables

rloterh committed:
Secure api key in saparate file

rloterh committed:
Create celcius to fahrenheit converter function

rloterh committed:
Create weather module

rloterh committed:
Add search event-listeners

rloterh committed:
Update styling

rloterh committed:
Update styling

rloterh committed:
Update ReadMe

rloterh committed:
Clean code

rloterh committed:
Clean code

rloterh committed:
Update ReadMe

MkrtichSargsyan reviewed (changes requested):
## _Status:_ Changes Requested :recycle:

## Hello Robert Loterh

THIS IS VERY IMPRESSIVE WORK. YOU HAVE DONE A GOOD JOB :+1:

## To Highlight :tada:

1. Your PR has a good title and description.
2. You have descriptive git commit messages.
3. Data is being fetched from the weather API correctly


However, there are some changes that you need to make in order to get your project approved

- [ ] Make development branch the default branch

- [ ] Kindly add a toggle control to determine if the result will be shown in Celsius or Fahrenheit [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/weather_app/search_requirement.png)

- [ ] Your app should change its appearance based on the received data [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/weather_app/appearance_requirement.png)

- [ ] It should have error handling for promises or async/await [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/weather_app/error_handling_requirement.png)

[Error handling with promises](https://javascript.info/promise-error-handling)



<hr>

## OPTIONAL

- [ ] It would be nice if you add a ‘loading’ component that displays from the time the form is submitted until the information comes back from the API

- [ ] It would be better if you fetch the current geolocation of the user when the app is first accessed and fetch the weather from that location


Please, make the REQUIRED changes and submit your request again! :grinning:
Good luck. 

> ps: . If you have any questions about my review or need help, you can message me on slack ( @mkrtichsargsyan )

MkrtichSargsyan commented on src/index.js:
```diff
@@ -0,0 +1,53 @@
+const weather = {
+  apiKey: weatherKey,
+  getWeatherData(city) {
+    fetch(
+      `https://api.openweathermap.org/data/2.5/weather?q=${
+        city
+      }&units=metric&appid=${
+        this.apiKey}`,
+    )
+      .then((response) => {
+        if (!response.ok) {
+          alert('No weather found.');
+          throw new Error('No weather found.');
+        }
+        return response.json();
+      })
+      .then((data) => this.displayWeather(data));
```
Add try-catch block for fetching data

rloterh committed:
Update ReadMe

rloterh committed:
Add Celsius and Fahrenheit conversion buttons

rloterh committed:
Add Celsius and Fahrenheit to element component

rloterh committed:
Update styling for Celsius and Fahrenheit buttons

rloterh committed:
Add event-listeners to Celsius and Fahrenheit conversion buttons

rloterh committed:
Refactor background image change on city search module

rloterh committed:
Update ReadMe

rloterh committed:
Update error handler

rloterh committed:
Clean code

rloterh commented:
Implemented recommended changes required (Please note, I intend to implement the optional changes later):

- Made the development branch the default branch.

 - Added a toggle control to determine if the result will be shown in Celsius or Fahrenheit.

 - App changes its appearance based on the received data.

  - It has error handling for promises or async/await

rloterh committed:
Clean code

rloterh commented:
Implemented recommended changes required:
(Please note, I intend to implement the optional changes later, thanks)

- Made the development branch the default branch.

 - Added a toggle control to determine if the result will be shown in Celsius or Fahrenheit.

 - App changes its appearance based on the received data.

  - It has error handling for promises or async/await

oloomoses reviewed (approved):
# Congratulations, Project Approved. :muscle: :tada: :clap: 
**Hello @rloterh,
Great job working on the changes suggested by @MkrtichSargsyan.
You have passed all the requirements for this project :muscle: 
Feel free to merge and close this pull request.**

## Optional
- [ ] Before merging this pr. Kindly delete the commented code in the `index.js` file. Commented code makes your work look messy.

**I wish you the best in your next project and see you at the top :rocket:** 
#### Happy Coding
_Slack: Oloo Moses_

rloterh committed:
Clean code

rloterh commented:
Noted. Thanks @oloomoses
